agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/statsbeat/NetworkStatsbeat.java

```diff
@@ -213,4 +238,13 @@ private double getRequestDurationAvg() {
       return sum;
     }
   }
+
+  /**
+   * e.g. endpointUrl 'https://westus-0.in.applicationinsights.azure.com/v2.1/track' host will
+   * return 'westus-0.in.applicationinsights.azure.com'
+   */
+  static String getHost(String endpointUrl) {
+    assert (endpointUrl != null && !endpointUrl.isEmpty());
+    return endpointUrl.replaceAll("^\\w+://", "").replaceAll("/\\w+.?\\w?/\\w+", "");
```

Comment:
instead of regex, can we use simpler indexOf("://") and indexOf("/", ...) and get substring in between those?